core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -65,8 +65,14 @@ class SocketServer(val config: KafkaConfig, val metrics: Metrics, val time: Time
   private val memoryPoolDepletedTimeMetricName = metrics.metricName("MemoryPoolDepletedTimeTotal", "socket-server-metrics")
   memoryPoolSensor.add(new Meter(TimeUnit.MILLISECONDS, memoryPoolDepletedPercentMetricName, memoryPoolDepletedTimeMetricName))
   private val memoryPool = if (config.queuedMaxBytes > 0) new SimpleMemoryPool(config.queuedMaxBytes, config.socketRequestMaxBytes, false, memoryPoolSensor) else MemoryPool.NONE
-  val requestChannel = new RequestChannel(maxQueuedRequests)
-  private val processors = new ConcurrentHashMap[Int, Processor]()
+  val dataRequestChannel = new RequestChannel(maxQueuedRequests, RequestChannel.RequestQueueSizeMetric)
+  var controlPlaneRequestChannel: RequestChannel = null
+  if (config.controlPlaneListenerName.isDefined) {
+    controlPlaneRequestChannel = new RequestChannel(20, RequestChannel.ControlPlaneRequestQueueSizeMetric)
+  }
+  private val dataProcessors = new ConcurrentHashMap[Int, Processor]()
+  // there should be only one controller processor, however we use a map to store it so that we can reuse the logic for data processors
+  private[network] val controlPlaneProcessors = new ConcurrentHashMap[Int, Processor]()
```

Comment:
Instead of having a map, which might indicate that we might have multiple processors for the controlPlane, do you think we can do : 
```suggestion
  private[network] var controlPlaneProcessorOpt : Option[Processor] = None
```
We can lazily create controlPlaneProcessor when we call startup()